src/gluonts/model/deepar/_network.py

```diff
@@ -274,6 +279,13 @@ def unroll_encoder(
             length=subsequences_length,
             layout="NTC",
             merge_outputs=True,
+            begin_state=self.rnn.begin_state(
```

Comment:
The default `begin_state` still uses `np.float32` even if the RNN cell is of type `np.float64`.

Reply:
Is this for hybrid sequential only, or also for elementary cells?

Reply:
My guess is that any dtype a HybridSequential(RNN) block deals with (dtype of output, dtype of state) depends on the dtypes its component blocks deal with.

Reply:
Yes, I remember similar issues with RNNs when doing this for the GP forecaster.

Reply:
I tested by casting each cell to proper `dtype` but that did not help either (see the comment below).